### PR TITLE
Add com.vscodium.codium

### PR DESCRIPTION
**Pull request by noonsleeper**
This repo is heavily based on Visual Studio Code flatpak repo
https://github.com/flathub/com.visualstudio.code

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "only-arches": ["x86_64"],
+  "skip-appstream-check": true
+}
```
Comment (LeandroStanger) [inline flathub.json]:
```suggestion
{
  "only-arches": ["x86_64"],
}
```

File: com.vscodium.codium.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      - type: file
+        url: https://raw.githubusercontent.com/VSCodium/vscodium/master/src/src/vs/workbench/contrib/update/browser/media/code-icon.svg
+        sha256: 4b0b8f198fda18a79bfd293f6ef7ba3db167f35324c97cbafda881840be8a981
+        dest-filename: codium.svg
+      - type: file
+        path: flatpak-warning.txt
+      - type: file
+        path: npmrc
+      - type: file
+        path: com.vscodium.codium.appdata.xml
```
Comment (LeandroStanger) [inline com.vscodium.codium.yaml]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Comment (noonsleeper) [reply com.vscodium.codium.yaml]: you mean I need to rename com.vscodium.codium.appdata.xml to com.vscodium.codium.metainfo.xml?

Comment (LeandroStanger) [reply com.vscodium.codium.yaml]: Yes

**Comment by LeandroStanger** [comment]
bot, build com.vscodium.codium

**Comment by nedrichards** [comment]
How exactly is this different to the OSS VS Code build https://github.com/flathub/com.visualstudio.code.oss ?

**Comment by noonsleeper** [comment]
> How exactly is this different to the OSS VS Code build https://github.com/flathub/com.visualstudio.code.oss ?

The code on https://github.com/flathub/com.visualstudio.code.oss is a build from vscode sources and sadly I don't have enough spare time to figured how to resolve any compilation time bug also https://github.com/flathub/com.visualstudio.code.oss/issues/46 is up to 7 month open. 
I only repackage vscodium.com's deb in the same way as flatpak's vscode build is made to take advantage of all the good things on flatpak.

**Comment by noonsleeper** [comment]
hey guys any news about this?

**Comment by bilelmoussaoui** [comment]
Have you reached out to the upstream maintainer if they wanted to (co-)maintain the flatpak package?

**Comment by noonsleeper** [comment]
hi! if you're asking me for the vscodium maintainer permissions I have it =)

**Comment by stripedpajamas** [comment]
Maintainer of VSCodium here-- I don't have capacity to maintain the flatpak version myself, but would be happy to help @noonsleeper etc whenever/however.

File: com.vscodium.codium.desktop [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[Desktop Entry]
+Name=VSCodium
+Comment=Code Editing. Redefined.
```
Comment (bilelmoussaoui) [inline com.vscodium.codium.desktop:3]: The desktop file & metainfo should be ideally upstreamed.

File: com.vscodium.codium.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
+app-id: com.vscodium.codium
+default-branch: stable
```
Comment (bilelmoussaoui) [inline com.vscodium.codium.yaml:2]:
```suggestion
```
it's stable by default on flathub.

Comment (noonsleeper) [reply com.vscodium.codium.yaml:2]: thanks!

File: com.vscodium.codium.desktop (hunk 3 above)
Comment (noonsleeper) [reply com.vscodium.codium.desktop:3]: @bilelmoussaoui, thanks for the review. I'll suggest this to the @stripedpajamas

**Comment by bilelmoussaoui** [comment]
bot, build com.vscodium.codium

File: com.vscodium.codium.desktop [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+GenericName=Text Editor
+Exec=codium --no-sandbox --unity-launch --reuse-window %F
```
Comment (bilelmoussaoui) [inline com.vscodium.codium.desktop:5]: You should use zypak instead of --no-sandbox.

Comment (barthalion) [reply com.vscodium.codium.desktop:5]: Note it's already included in the base app so you just need to change this line to use zypak-wrapper. Example app: https://github.com/flathub/com.microsoft.Teams

Comment (noonsleeper) [reply com.vscodium.codium.desktop]: hey! sorry for the late response recently at my job the amount work has increased. @bilelmoussaoui I applied your suggestion, thanks for the link @barthalion. Also I update package to the latest version of the Codium.

**Comment by barthalion** [comment]
bot, build com.vscodium.codium

File: com.vscodium.codium.yaml [hunk 6]
```diff
@@ -0,0 +1,82 @@
+app-id: com.vscodium.codium
+base: org.electronjs.Electron2.BaseApp
+base-version: '19.08'
+runtime: org.freedesktop.Sdk
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+command: codium
+desktop-file-name-suffix: ' (Flatpak)'
```
Comment (barthalion) [inline com.vscodium.codium.yaml:8]: I understand you want to differentiate it but I don't think it's the way to go.

Comment (noonsleeper) [reply com.vscodium.codium.yaml:8]: ohh no! I miss that change, let me fix that.

Comment (noonsleeper) [reply com.vscodium.codium.yaml:8]: > Have you contacted upstream team about publishing their application on Flathub?

Yes, @stripedpajamas is the developer of VSCodium 

> Maintainer of VSCodium here-- I don't have capacity to maintain the flatpak version myself, but would be happy to help @noonsleeper etc whenever/however.

**Comment by barthalion** [comment]
FYI @stripedpajamas, I'm going to add you to maintainers as well, just in case you want to change something yourself.

/merge @stripedpajamas

**Pull request closed without merging**